protocol/messenger.go

```diff
@@ -2043,24 +2044,60 @@ func (m *Messenger) Mailservers() ([]string, error) {
 	return nil, ErrNotImplemented
 }
 
-func (m *Messenger) initChatFirstMessageTimestamp(chat *Chat) error {
-	if !chat.CommunityChat() || chat.FirstMessageTimestamp != FirstMessageTimestampUndefined {
-		return nil
+func (m *Messenger) initChatsFirstMessageTimestamp(chats []*Chat) {
+	var communityChats []*Chat
+	var communityChatIDs []string
+	for _, chat := range chats {
+		if !chat.CommunityChat() || chat.FirstMessageTimestamp != FirstMessageTimestampUndefined {
+			continue
+		}
+		communityChats = append(communityChats, chat)
+		communityChatIDs = append(communityChatIDs, chat.ID)
+	}
+	if len(communityChatIDs) == 0 {
+		return
 	}
 
-	oldestMessageTimestamp, hasAnyMessage, err := m.persistence.OldestMessageWhisperTimestampByChatID(chat.ID)
+	oldestMessageTimestamps, err := m.persistence.OldestMessageWhisperTimestampByChatIDs(communityChatIDs)
 	if err != nil {
-		return err
+		m.logger.Warn("failed to get oldest message timestamp", zap.Error(err))
+		return
 	}
+	communityCache := make(map[string]*communities.Community)
```

Comment:
We already have the same type of cache done above called `communityInfo`. Could we move this code to the main function to reuse the cache, or pass a ref to the cache, or even move the community cache to a `Messenger` prop?

I think passing a ref is probably the simplest and cleanest